### PR TITLE
test(datetime): event for calendar month text change

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Flaky e2e tests

## What is the current behavior?

The calendar month text updating can take longer than the hard set 350ms timeout in the e2e tests.

Issue URL: 

## What is the new behavior?

Using a mutation observer, we can dispatch a custom event that the test can listen for when the text is actually modified.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/datetime/test/sub-pixel-width/index.html [hunk 1]
```diff
@@ -42,6 +42,25 @@ <h2>Modal</h2>
       </ion-modal>
     </ion-content>
   </ion-app>
+
+  <script>
+    const openModal = document.getElementById('open-modal');
+
+    const observer = new MutationObserver(mutationRecords => {
```
Comment (amandaesmith3) [inline core/src/components/datetime/test/sub-pixel-width/index.html:49]: Could we pull this to a util file somewhere so other tests can import it? (And also make the event name more generic?)

File: core/src/components/datetime/test/sub-pixel-width/index.html [hunk 2]
```diff
@@ -42,6 +42,25 @@ <h2>Modal</h2>
...
+      if (mutationRecords[0].type === 'characterData') {
+        document.dispatchEvent(new CustomEvent('datetimeCalendarMonthSpecTestEvent'));
+      }
+    });
+
+    openModal.addEventListener('click', () => {
+      observer.observe(document.querySelector('ion-datetime').shadowRoot.querySelector('.calendar-month-year'), {
+        childList: true,
```
Comment (amandaesmith3) [inline core/src/components/datetime/test/sub-pixel-width/index.html:57]: Is `childList` needed here if we're just checking text content? I do remember there being some cross-browser jank with observing that, so I'm not actually sure 🤔

**Review by amandaesmith3 — CHANGES_REQUESTED** [review]

File: core/src/components/datetime/test/sub-pixel-width/index.html (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/datetime/test/sub-pixel-width/index.html]: Updated, let me know if this aligns with what you were thinking 👍

File: core/src/components/datetime/test/sub-pixel-width/index.html (hunk 2 above)
Comment (sean-perkins) [reply core/src/components/datetime/test/sub-pixel-width/index.html]: Good catch, that piece wasn't required for listening to slot changes. Removed!

File: core/src/components/datetime/test/sub-pixel-width/index.html (hunk 1 above)
Comment (amandaesmith3) [reply core/src/components/datetime/test/sub-pixel-width/index.html]: Looks great, thanks!

**Review by amandaesmith3 — APPROVED** [review]

**Pull request merged**